R/keep.R

```diff
@@ -58,3 +59,29 @@ compact <- function(.x, .p = identity) {
   .f <- as_mapper(.p)
   discard(.x, function(x) is_empty(.f(x)))
 }
+
+
+#' Keep/discard elements based on their name/position
+#'
+#' @inheritParams map_at
+#' @seealso [keep()]/[discard()] to keep/discard elements by value.
+#' @export
+#' @examples
+#' x <- c(a = 1, b = 2, cat = 10, dog = 15, elephant = 5, e = 10)
+#' x %>% keep_at(letters)
+#' x %>% discard_at(letters)
+#'
+#' # Can also use a function
+#' x %>% keep_at(~ nchar(.x) == 3)
+#' x %>% discard_at(~ nchar(.x) == 3)
+keep_at <- function(x, at) {
+  where <- at_selection(x, at)
+  x[where]
+}
+
+#' @export
+#' @rdname keep_at
+discard_at <- function(x, at) {
+  where <- at_selection(x, at)
+  x[-where]
```

Comment:
I don't think you can use `-where` because it has issues when you select nothing

``` r
discard_at(c(x = 1), "y")
#> named numeric(0)
```

I think you could maybe use `where <- inv_which(x, where); x[where]`? At first glance that helper looks right